src/moin/storage/middleware/indexing.py

```diff
@@ -1233,7 +1233,8 @@ def store_revision(self, meta, data, overwrite=False,
         backend = self.backend
         if not overwrite:
             revid = meta.get(REVID)
-            if revid is not None and revid in backend:
+            backend_name = dict(app.cfg.namespace_mapping)[meta.get(NAMESPACE, '')]
+            if revid is not None and (revid in backend or (backend_name, revid) in backend):
```

Comment:
while developing the test this presented itself as a bug

 when SetMeta was still lacking the `del meta[REVID]` I did not get the expected ValueError until making this change